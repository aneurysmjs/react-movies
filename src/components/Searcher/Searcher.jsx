import React from 'react';

export default class Searcher extends React.Component {

  constructor(props) {
    super(props);

    // This binding is necessary to make `this` work in the callback
    this.search = this.search.bind(this);
  }

  render() {
    return (
      <div className='row justify-content-center'>
        <div className='col-xs-12 col-sm-4 col-md-4'>
          <h5 className='text-center'>Search Movies</h5>
          <form name='searcher' onSubmit={this.search}>
            <div className='form-group'>
              <label htmlFor='searcher' className='sr-only'>Just type</label>
              <input
                type='text'
                id='searcher'
                className='form-control'
                name='searcher'
                placeholder='Type...'
                value={this.props.searchTerm}
                onChange={this.search} />
            </div>
          </form>
        </div>
      </div>
    );
  }

  /**
   * pass data to parent component
   *
   * @param {SyntheticEvent} evt - the event that comes from the input
   * @return {void}
   */
  search(evt) {
    this.props.onSearch(evt.target.value);
  }
  
}