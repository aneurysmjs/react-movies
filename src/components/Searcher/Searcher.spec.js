import React from 'react';
import renderer from 'react-test-renderer';
import Searcher from './Searcher';

test('Searcher snapshot test', () => {

  const component = renderer.create(<Searcher />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();

});