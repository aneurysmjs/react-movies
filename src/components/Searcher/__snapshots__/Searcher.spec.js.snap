// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Searcher snapshot test 1`] = `
<div
  className="row justify-content-center"
>
  <div
    className="col-xs-12 col-sm-4 col-md-4"
  >
    <h5
      className="text-center"
    >
      Search Movies
    </h5>
    <form
      name="searcher"
      onSubmit={[Function]}
    >
      <div
        className="form-group"
      >
        <label
          className="sr-only"
          htmlFor="searcher"
        >
          Just type
        </label>
        <input
          className="form-control"
          id="searcher"
          name="searcher"
          onChange={[Function]}
          placeholder="Type..."
          type="text"
          value={undefined}
        />
      </div>
    </form>
  </div>
</div>
`;
